x-pack/plugins/lens/public/datasources/form_based/form_based.tsx

```diff
@@ -970,18 +970,40 @@ export function getFormBasedDatasource({
       return Object.values(state.layers).map(({ indexPatternId }) => indexPatternId);
     },
 
-    getDatasourceInfo: (state, references, indexPatterns) => {
+    getDatasourceInfo: async (state, references, dataViewsService) => {
       const layers = references ? injectReferences(state, references).layers : state.layers;
+      const indexPatterns: DataView[] = [];
+      for (const { indexPatternId } of Object.values(layers)) {
+        const dataView = await dataViewsService?.get(indexPatternId);
+        if (dataView) {
+          indexPatterns.push(dataView);
+        }
+      }
```

Comment:
Potentially this can be very slow for multiple layers: can we fetch them concurrently?
Also, it could skip duplicates.

Reply:
Can we reuse the initialization code for indexPatterns loading within the `ChartInfoAPI`? That optimizes already the amount of requests.

Reply:
@dej611 As I know `dataViewsService.get` have the caching mechanism, because of this I don't see any problem with performance